src/reformatters/example/new_dataset.py

```diff
@@ -0,0 +1,94 @@
+import re
+import shutil
+from pathlib import Path
+from typing import Annotated
+
+import typer
+
+
+def initialize_new_integration(
+    provider: Annotated[
+        str, typer.Argument(help="The provider name in lowercase (e.g. 'noaa')")
+    ],
+    model: Annotated[
+        str, typer.Argument(help="The model name in lowercase (e.g. 'gfs')")
+    ],
+    variant: Annotated[
+        str, typer.Argument(help="The variant name in lowercase (e.g. 'forecast')")
+    ],
+) -> None:
+    """Create a new dataset integration from the example template."""
+    # Sanitize inputs
+    provider = _sanitize_identifier(provider)
+    model = _sanitize_identifier(model)
+    variant = _sanitize_identifier(variant)
+
+    # Convert to PascalCase for class names
+    provider_pascal = provider.capitalize()
+    model_pascal = model.capitalize()
+    variant_pascal = variant.capitalize()
+
+    # Set up paths
+    dataset_path = f"{provider}/{model}/{variant}"
+    src_path = Path("src/reformatters") / dataset_path
+    test_path = Path("tests") / dataset_path
+
+    # Create directories
+    src_path.mkdir(parents=True, exist_ok=True)
+    test_path.mkdir(parents=True, exist_ok=True)
+
+    # Copy example files
+    example_src = Path("src/reformatters/example")
+    example_test = Path("tests/example")
+
+    for file in example_src.glob("*"):
+        if file.is_file() and file.name != "new_dataset.py":  # Skip this file
+            shutil.copy(file, src_path / file.name)
+
+    for file in example_test.glob("*"):
+        if file.is_file():
+            shutil.copy(file, test_path / file.name)
+
+    # Create empty __init__.py
+    (src_path / "__init__.py").touch()
+    (test_path / "__init__.py").touch()
+
+    # Perform renames in copied files
+    example_to_actual_mappings = {
+        "ExampleDataset": f"{provider_pascal}{model_pascal}{variant_pascal}Dataset",
+        "ExampleTemplateConfig": f"{provider_pascal}{model_pascal}{variant_pascal}TemplateConfig",
+        "ExampleRegionJob": f"{provider_pascal}{model_pascal}{variant_pascal}RegionJob",
+        "ExampleDataVar": f"{provider_pascal}{model_pascal}DataVar",
+        "ExampleInternalAttrs": f"{provider_pascal}{model_pascal}InternalAttrs",
+        "ExampleSourceFileCoord": f"{provider_pascal}{model_pascal}SourceFileCoord",
```

Comment:
I realized because the `.out_loc` is often different per variant we want the variant in the class name too

```suggestion
        "ExampleSourceFileCoord": f"{provider_pascal}{model_pascal}{variant_pascal}SourceFileCoord",
```

Reply:
oo nice good catch, updating